### DISCUSSION TITLE
Different message templates per delivery method

### DISCUSSION THREAD

**Discussion by howdymatt** [excid3/noticed#269] [Q&A]
I have a use case of sending a message via email or SMS. We would prefer to use ERB templates for both delivery methods, but the email template would include more text / styling than the SMS would.

I would also like the templates to reside somewhere that makes sense, like `views/notifications/my-message-type.delivery-method.erb` or better.

What would be the best way to begin this? I'm having trouble decide which class should be responsible for the _building_ of the message, from ERB to string (with i18n), for each message type.

**Comment by excid3** [comment]
You can supply the option to the delivery method to customize what they send so Twilio can have a different message than another one. https://github.com/excid3/noticed/blob/master/docs/delivery_methods/twilio.md#options

Emails will use the Mailer views so you can customize those templates like normal Rails emails.

**Reply by howdymatt** [reply]
Thank you for the fast response!

I apologize, I didn't mention that we are using our own service for sending email and SMS and cannot rely on Twilio and Mailer (for specific app needs). I like the Mailer options for templating, but since we aren't using it for SMS I'll be rolling my own service to call, with separate custom delivery methods for email and sms.

**Reply by excid3** [reply]
Sure, but the point is that you can pass in arguments to each delivery method, so you can use that to customize the messages for each. You'll use that same approach for your custom delivery methods.
